zap/src/main/java/org/parosproxy/paros/control/Control.java

```diff
@@ -666,4 +668,33 @@ public void setMode(Mode mode) {
             model.getOptionsParam().getViewParam().setMode(mode.name());
         }
     }
+
+    public static LinkedList<String> getRecentSessionList() {
+        LinkedList<String> recentSessionList = new LinkedList<>();
+        for (String path :
+                Model.getSingleton()
+                        .getOptionsParam()
+                        .getConfig()
+                        .getStringArray("recentsessions.path")) recentSessionList.add(path);
+        return recentSessionList;
+    }
+
+    private void setRecentSessionList(String path) {
+        LinkedList<String> recentSessionList = getRecentSessionList();
+        int index = recentSessionList.indexOf(path);
+        if (index == 0) return;
+        else if (index > 0) recentSessionList.remove(index);
+        recentSessionList.addFirst(path);
+        while (recentSessionList.size() > 10) recentSessionList.removeLast();
+        try {
+            FileConfiguration config = model.getOptionsParam().getConfig();
+            config.clearProperty("recentsessions.path");
+            for (int i = 0; i < recentSessionList.size(); ++i)
+                config.setProperty("recentsessions.path(" + i + ")", recentSessionList.get(i));
+            config.save();
+            View.getSingleton().getMainFrame().getMainMenuBar().setMenuFileOpenRecent();
```

Comment:
Always check that the View has been initialised before trying to access it - its wont have been in daemon or command line modes. Unless it will only ever be called from GUI code but I dont think thats the case here.